docs/docs/language_reference/schema/index.md

```diff
@@ -225,13 +226,26 @@ type User {
 }
 ```
 
-Here we've split properties and edges and used non-doc comments such as `# own properties`
-to signal this. Also, as a user can make multiple posts and follow multiple people the type
+For readability, here we've split properties and edges using non-doc comments such as `# own edges`.
+
+As a user can make multiple posts and follow multiple people, the type
 of both edges is a list. Posts is a list of non-null post objects with the type `[Post!]` and
 follows is similar a list of non-null users `[User!]`.
 
-Applying a similar change to Post we now add an edge for the author of the post, a single non-null
-User.
+There are some nullability rules to be aware of with the Trustfall edges, these are all the valid
+forms of a `Post` edge: 
+
+* `Post` is 0 or 1 post
+* `Post!` is exactly 1 post
+* [Post!] is 0 or more posts
+* [Post!]! is 1 or more posts
+
+In our example the last one is the only one we won't be using. By making the list non-null it
+guarantees a single value. We can't place just a `Post` as having a list of multiple neighbours 
+that don't exist doesn't make sense — and would also complicate our queries.
```

Comment:
I'm confused about this:
> By making the list non-null it guarantees a single value.

Also the next sentence seems not entirely grammatical. Perhaps a small rewrite?

Reply:
okay gave it another shot